### PR TITLE
Introduce not given and apply it to dataset's target

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Description

Introduce not given and apply it to dataset's target

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [x] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

**Review by rabah-khalek — CHANGES_REQUESTED** [review]
Looks good to me @kevinmessiaen.

Could we also undo the changes done in https://github.com/Giskard-AI/giskard/pull/1531/files (the ones related to suppress the warning of target for LLM) and add the `target=None` in the BaseModel for when `model_type == "text_generation"` to suppress the warning?

**Comment by kevinmessiaen** [comment]
> Looks good to me @kevinmessiaen.
> 
> Could we also undo the changes done in https://github.com/Giskard-AI/giskard/pull/1531/files (the ones related to suppress the warning of target for LLM) and add the `target=None` in the BaseModel for when `model_type == "text_generation"` to suppress the warning?

Not sure to understand the logic of the validation though, why to we validate the dataset target when validating model?

For me this validation is completely unrelated to the model validation and should be done when validating dataset only.
For me when uploading a model we should not confuse the user with warning that the dataset is missing a target, since it's unrelated.

**Comment by rabah-khalek** [comment]
that's exactly what I meant by my comment. We should remove it. 
We had to do it to suppress the warning only for LLMs. We needed to do it inside validate_model just to know the model_type back then. But now people can put explicitly `target=None` to supress this warning

**Comment by kevinmessiaen** [comment]
> that's exactly what I meant by my comment. We should remove it. We had to do it to suppress the warning only for LLMs. We needed to do it inside validate_model just to know the model_type back then. But now people can put explicitly `target=None` to supress this warning

Thanks for clarification, so it's what I did then, you can review the PR when you have time

File: giskard/core/model_validation.py [hunk 1]
```diff
@@ -12,15 +12,15 @@
...
 from ..utils.analytics_collector import analytics, get_dataset_properties, get_model_properties
-from .dataset_validation import validate_optional_target
 
 
 @configured_validate_arguments
 def validate_model(model: BaseModel, validate_ds: Optional[Dataset] = None, print_validation_message: bool = True):
     try:
-        if model.meta.model_type != SupportedModelTypes.TEXT_GENERATION and validate_ds is not None:
+        if validate_ds is not None:
             validate_optional_target(validate_ds)
```
Comment (rabah-khalek) [inline giskard/core/model_validation.py:24]: what I mean is that we should restore this to dataset validation, and put target=None in the constructor of the BaseModel if model_type=="text_generation".

Let me know if it's unclear or if you're not convinced, we can jump on a quick call.

**Review by rabah-khalek — APPROVED** [review]
Thanks @kevinmessiaen, looks good

**Pull request merged**